### ISSUE TITLE
[BUG] {{random}} inside {{random}}

### ISSUE THREAD

**Issue by AnonCardMaker** [SillyTavern/SillyTavern#882]
**Have you searched for similar [bugs](https://github.com/SillyTavern/SillyTavern/issues?q=)?**
Yes

**Describe the bug**
If you put {{random}} inside another {{random}} it just make a mess.
Some results from "{{random:red,{{random:green,blue}}, yellow}}":
yellow
blue,yellow}}
red, yellow}}

**To Reproduce**
Copypaste {{random:red,{{random:green,blue}}, yellow}}

**Expected behavior**
The formula {{random:red,{{random:green,blue}}, yellow}} should produce "red" 33%, green 16.5%, blue 16.5% and "yellow" 33%, of the time.

**Comment by Cohee1207** [comment]
Nested randoms are not supported, as not as easily achievable with regex alone

**Comment by AeonBlack** [comment]
Weighted randoms may be a better solution, {{random:red:2,green,blue,yellow:2}} should be achievable with default weight as 1 where colon is missing.

**Comment by AnonCardMaker** [comment]
> Weighted randoms may be a better solution, {{random:red:2,green,blue,yellow:2}} should be achievable with default weight as 1 where colon is missing.

I'd really appreciate it, although it still wouldn't work in a situation like {{random:{{user}} finds a {{random:weapon,potion,scroll}},{{user}} is attacked by a {{random:goblin,wolf,orc}}}} (I guess in that example you could manually repeat each possibility)


> Nested randoms are not supported, as not as easily achievable with regex alone

Ah, a shame. Still would like that as a feature. Maybe in the future.

**Comment by drago87** [comment]
Weighted randoms would be really nice to have

> Weighted randoms may be a better solution, {{random:red:2,green,blue,yellow:2}} should be achievable with default weight as 1 where colon is missing.

**Comment by Wolfsblvt** [comment]
Yeah, Macro 2.0 ™️  is still in the works, and will be finished some time soon ™️.

**Comment by underscorex86** [comment]
soon

**Comment by Silly-Jarv** [comment]
I hope this request is still relevant and will be given consideration.

It is also possible in STScript with an /if, but it is very time-consuming to write.

This feature would be a great help, or with weights. Either would be great.

**Comment by Wolfsblvt** [comment]
Soon is **now**. Or like.. a few weeks ago.

Has been implemented and can be tested on **staging**.
See #4820 for more details.

The new macro engine currently still needs to be enabled:
<img width="582" height="142" alt="Image" src="https://github.com/user-attachments/assets/8d44f66f-26df-4635-a29c-9a9baa2e22aa" />

**Comment by Cohee1207** [comment]
Update to 1.15.0 and enable "Experimental Macro Engine" in User Settings.
